src/main/kotlin/telegram/Menu.kt

```diff
@@ -195,124 +284,375 @@ internal fun showRevealMenu(game: Game, bot: Bot, chatId: Long, messageId: Long)
+        "Ведущие",
+        hostSettings,
+        {
+            button(chooseHostAdminCommand named (it.host?.fullName()?: ""), -1L, it.hostId)
+        },
+        hostAdminSettingsCommand,
+        pageIndex
     )
 }
 
-internal fun showChosenSettingsMenu(chatId: Long, messageId: Long, bot: Bot, chosenId: Long) {
-    hostSettings.get(chosenId)?.let { settings ->
+fun sendMessage(
+    bot: Bot,
+    chatId: Long,
+    text: String,
+    replyMarkup: (Long) -> ReplyMarkup? = { null },
+): Long {
+    val res = bot.sendMessage(
+        ChatId.fromId(chatId),
+        text,
+        parseMode = ParseMode.HTML
+    )
+    return if (res.isSuccess) {
+        val msgId = res.get().messageId
+        updateMessage(
+            bot,
+            chatId,
+            msgId,
+            replyMarkup = replyMarkup(msgId)
+        )
+        msgId
+    } else {
+        -1L
+    }
+}
+
+fun updateMessage(
+    bot: Bot,
+    chatId: Long,
+    messageId: Long,
+    text: String? = null,
+    replyMarkup: ReplyMarkup? = inlineKeyboard {  }
+) {
+    if (text == null) {
         bot.editMessageReplyMarkup(
             ChatId.fromId(chatId),
             messageId,
+            replyMarkup = replyMarkup
+        )
+    } else {
+        bot.editMessageText(
+            ChatId.fromId(chatId),
+            messageId,
+            text = text,
+            parseMode = ParseMode.HTML,
+            replyMarkup = replyMarkup
+        )
+    }
+}
+
+internal fun showChosenSettingsMenu(chatId: Long, messageId: Long, bot: Bot, chosenId: Long) {
+    hostSettings.get(chosenId)?.let { settings ->
+        val msgId = if (messageId == -1L) {
+            sendMessage(bot, chatId, "Настройки ведущего")
+        } else {
+            messageId
+        }
+        updateMessage(
+            bot,
+            chatId,
+            msgId,
             replyMarkup = inlineKeyboard {
-                button(blankCommand named "Настройки ${accounts.get(chosenId)?.fullName()?: ""}")
+                button(blankCommand named "Настройки ${accounts.get(chosenId)?.fullName() ?: ""}")
                 HostOptions.entries.forEach { entry ->
                     row {
-                        button(changeHostAdminSettingCommand named entry.shortName, messageId, chosenId, entry.name)
-                        button(changeHostAdminSettingCommand named (if (entry.current(settings)) "✅" else "❌"), messageId, chosenId, entry.name)
+                        button(changeHostAdminSettingCommand named entry.shortName, msgId, chosenId, entry.name)
+                        button(
+                            changeHostAdminSettingCommand named (if (entry.current(settings)) "✅" else "❌"),
+                            msgId,
+                            chosenId,
+                            entry.name
+                        )
                     }
                 }
-                button(adminBackCommand, messageId)
+                button(deleteMsgCommand named "Закрыть", msgId)
```

Comment:
Same problem as with sending a message at the beginning of this function. You are changing business logic without a reason and even without necessary updates at the call site